### PR TITLE
Feature/prml 502 boolean features are currently not handled in redshift and Snowflake

### PR DESCRIPTION
**Pull request by chetana1410**
Modified the code in all 3 warehouses such that boolean features are handled

### CODE REVIEW

File: samples/predictions_dev_project/models/profiles.yaml [hunk 1]
```diff
@@ -27,7 +27,14 @@ models:
...
               - max_timestamp_bw_tracks_pages
-
+        - entity_var:
+            name: is_churned_7_days
+            select: case when {{user.Var("days_since_last_seen")}} > 7 then 1 else 0 end
+            description: Depending on the n value, it specifies if there is any activity observed in the last 7 days.
+        - entity_var:
+            name: random_boolean_feature
+            select: "{{macro_random_bool_feature()}}"
+            description: this feature is created for testing purpose. Remove it later.
```
Comment (dpatchigolla) [inline samples/predictions_dev_project/models/profiles.yaml:37]: `Remove it later.` - you can keep it, sample projects are just those - samples.

File: src/predictions/rudderstack_predictions/connectors/CommonWarehouseConnector.py [hunk 2]
```diff
@@ -111,6 +111,17 @@ def transform_arraytype_features(
...
+    def transform_booleantype_features(
+        self, feature_df: pd.DataFrame, booleantype_features: List[str]
+    ) -> Union[List[str], pd.DataFrame]:
+        transformed_feature_df = feature_df.copy()
+
+        for boolean_column in booleantype_features:
+            transformed_feature_df[boolean_column] = transformed_feature_df[
+                boolean_column
+            ].astype(int)
+        return booleantype_features, transformed_feature_df
```
Comment (dpatchigolla) [inline src/predictions/rudderstack_predictions/connectors/CommonWarehouseConnector.py:123]: Why do you need to return `boolean_type_features`? It's the input to the function, and it is not getting modified anyway in this function.

File: src/predictions/rudderstack_predictions/connectors/SnowflakeConnector.py [hunk 3]
```diff
@@ -410,6 +424,20 @@ def transform_arraytype_features(
...
+        """Transforms booleantype features in a snowflake.snowpark.Table"""
+
+        # Initialize a variable to store the original feature table
+        transformed_feature_table = feature_table
+
+        for boolean_column in booleantype_features:
+            transformed_feature_table = transformed_feature_table.withColumn(
+                boolean_column, F.col(boolean_column).cast("integer")
+            )
+        return booleantype_features, transformed_feature_table
```
Comment (dpatchigolla) [inline src/predictions/rudderstack_predictions/connectors/SnowflakeConnector.py:439]: Same comment as in other file. Why do you need to return `booleantype_features` again?

File: src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 
 from ...utils.logger import logger
 from .connector_base import ConnectorBase
+import time
```
Comment (dpatchigolla) [inline src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py:16]: Why is this import required?

File: src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py [hunk 5]
```diff
@@ -52,6 +53,7 @@ def write_to_table(
                     schema=schema,
                     index=False,
                     if_exists=if_exists,
+                    method="multi",
```
Comment (dpatchigolla) [inline src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py:56]: What does this do?

File: tests/unit/RedshiftConnector.py [hunk 6]
```diff
@@ -1,8 +1,10 @@
 # Generated by CodiumAI
 import unittest
+import unittest
```
Comment (dpatchigolla) [inline tests/unit/RedshiftConnector.py:3]: Duplicate import. Also namedtuple.

File: src/predictions/rudderstack_predictions/connectors/CommonWarehouseConnector.py (hunk 2 above)
Comment (dpatchigolla) [reply src/predictions/rudderstack_predictions/connectors/CommonWarehouseConnector.py:123]: Can you add a unittest for this function? 
Make sure following scenarios are tested:
1. Input column has only nulls
2. Input column has True, False, and a few nulls

**Review by dpatchigolla — COMMENTED** [review]
Added a few comments.

File: src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py (hunk 5 above)
Comment (chetana1410) [reply src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py:56]: When doing a pb run in redshift, idf.to_sql was taking a lot of time but it was not ending even after 3-4 hrs, method='multi' helped in speeding up a lot, was able to write within a minute

Comment (dpatchigolla) [reply src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py:55]: Are you writing through s3?

Comment (chetana1410) [reply src/predictions/rudderstack_predictions/connectors/wh/redshift_base_connector.py:55]: https://stackoverflow.com/a/54476922
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_sql.html

**Review by dpatchigolla — APPROVED** [review]

**Pull request merged**